Build fails when parsing an endpoint that has a bean annotated with @BeanParam when the target object is inheriting its fields from a parent class

### Describe the bug

When using version **2.6.2.Final**, the build fails when parsing an endpoint that has a bean annotated with @BeanParam, if the bean is inheriting its fields from a parent class:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12.850 s
[INFO] Finished at: 2022-01-17T15:37:28Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.6.2.Final:build (default) on project quarkus-resteasy-beanparam: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupEndpoints threw an exception: java.lang.RuntimeException: java.lang.RuntimeException: Failed to process method 'org.acme.rest.client.BeanParamResource#hello'
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:278)
[ERROR]         at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.setupEndpoints(ResteasyReactiveProcessor.java:503)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:887)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.lang.RuntimeException: Failed to process method 'org.acme.rest.client.BeanParamResource#hello'
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:609)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:336)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:249)
[ERROR]         ... 12 more
[ERROR] Caused by: javax.enterprise.inject.spi.DeploymentException: No annotations found on fields at 'org.acme.rest.client.MyParams'. Annotations like `@QueryParam` should be used in fields, not in methods.
[ERROR]         at org.jboss.resteasy.reactive.server.processor.ServerEndpointIndexer.handleBeanParam(ServerEndpointIndexer.java:171)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:493)
[ERROR]         ... 14 more
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

### Expected behavior

Project should compile and run regardless of inheritance on the annotated bean.

### Actual behavior

Project's build fails and no artifact is generated.

### How to Reproduce?

1. Clone the project: https://github.com/fernandobalieiro/quarkus-resteasy-beanparam
2. Run `./mvnw clean package` or `./mvnw clean install` or `./mvnw quarkus:dev`
3. Build fails and aforementioned Exception is shown.

### Output of `uname -a` or `ver`

Linux 5.11.0-46-generic #51~20.04.1-Ubuntu SMP Fri Jan 7 06:51:40 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment GraalVM CE 21.0.0.2 (build 11.0.10+8-jvmci-21.0-b06)
OpenJDK 64-Bit Server VM GraalVM CE 21.0.0.2 (build 11.0.10+8-jvmci-21.0-b06, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /home/fernandobalieiro/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4
Java version: 11.0.10, vendor: GraalVM Community, runtime: /home/fernandobalieiro/.asdf/installs/java/graalvm-21.0.0.2+java11
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.11.0-46-generic", arch: "amd64", family: "unix"
### Additional information

Reverting Quarkus to version **2.5.4.Final** solves the issue.

--- a/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java
+++ b/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java
@@ -167,7 +167,7 @@ public class ServerEndpointIndexer
         InjectableBean injectableBean = scanInjectableBean(beanParamClassInfo,
                 actualEndpointInfo,
                 existingConverters, additionalReaders, injectableBeans, hasRuntimeConverters);
-        if (injectableBean.getFieldExtractorsCount() == 0) {
+        if ((injectableBean.getFieldExtractorsCount() == 0) && !injectableBean.isInjectionRequired()) {
             throw new DeploymentException(String.format("No annotations found on fields at '%s'. "
                     + "Annotations like `@QueryParam` should be used in fields, not in methods.",
                     beanParamClassInfo.name()));
@@ -288,7 +288,7 @@ public class ServerEndpointIndexer
 
         currentInjectableBean.setFieldExtractorsCount(fieldExtractors.size());
 
-        if (!fieldExtractors.isEmpty() && fieldInjectionHandler != null) {
+        if ((fieldInjectionHandler != null) && (!fieldExtractors.isEmpty() || superTypeIsInjectable)) {
             fieldInjectionHandler.handleFieldInjection(currentTypeName, fieldExtractors, superTypeIsInjectable);
         }
         currentInjectableBean.setInjectionRequired(!fieldExtractors.isEmpty() || superTypeIsInjectable);

--- a/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/BeanParamEmptySubclassTest.java
+++ b/independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/simple/BeanParamEmptySubclassTest.java
@@ -1,0 +1,58 @@
+package org.jboss.resteasy.reactive.server.vertx.test.simple;
+
+import io.restassured.RestAssured;
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import org.hamcrest.Matchers;
+import org.jboss.resteasy.reactive.server.vertx.test.framework.ResteasyReactiveUnitTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class BeanParamEmptySubclassTest {
+
+    @RegisterExtension
+    static ResteasyReactiveUnitTest test = new ResteasyReactiveUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(TestResource.class, BaseParams.class, Params.class));
+
+    @Test
+    public void test() {
+        RestAssured.given().formParam("param1", "foo").post("/bean?param2=bar")
+                .then().statusCode(200).body(Matchers.equalTo("foo/bar"));
+    }
+
+    @Path("bean")
+    public static class TestResource {
+
+        @POST
+        @Produces(MediaType.TEXT_PLAIN)
+        public String hello(final @BeanParam BaseParams params) {
+            return params.getParam1() + "/" + params.getParam2();
+        }
+    }
+
+    public static class BaseParams {
+        @FormParam("param1")
+        private String param1;
+
+        @QueryParam("param2")
+        private String param2;
+
+        public String getParam1() {
+            return param1;
+        }
+
+        public String getParam2() {
+            return param2;
+        }
+    }
+
+    public static class Params extends BaseParams {
+
+    }
+}

